src/Dropbox.elm

```diff
@@ -861,7 +924,148 @@ upload auth info =
             ]
         , url = url
         , body = body
-        , expect = Http.expectJson decodeUploadResponse
+        , expect = Http.expectJson decodeFileMetadata
+        , timeout = Nothing
+        , withCredentials = False
+        }
+        |> Http.toTask
+        |> Task.mapError decodeError
+
+
+{-| Request parameters for `listFolder`
+-}
+type alias ListFolderRequest =
+    { path : String
+    , recursive : Bool
+    , includeMediaInfo : Bool
+    , includeDeleted : Bool
+    , includeHasExplicitSharedMembers : Bool
+    }
+
+
+{-| See <https://www.dropbox.com/developers/documentation/http/documentation#files-list_folder>
+-}
+type ListFolderError
+    = PathListError LookupError
+    | ExpiredCursorError
```

Comment:
I don't see `ExpiredCursorError` in the dropbox API docs (though it does say list_folder error is an open union?  Where does it come from?